src/collectors/ip/test/testip.py

```diff
@@ -38,49 +38,70 @@ def test_import(self):
     def test_should_open_proc_net_snmp(self, publish_mock, open_mock):
         IPCollector.PROC = ['/proc/net/snmp']
         open_mock.return_value = StringIO('')
-        self.collector.collect()
+        self.collector.collect_ipv4()
```

Comment:
I think it makes more send to call `self.collector.collect()` instead of the ipv4/ipv6 variants. I'm sure you can configure different mock return values based on the filename passed in.